Simplify tmpl::list_contains, add lazy version

wthrowe opened the pull request:
Use a fold expression instead of Brigand.  This does not appear to have a significant effect on compile time.

## Proposed changes

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

wthrowe committed:
Simplify tmpl::list_contains

wthrowe committed:
Reset compute items in test_reset_compute_items

wthrowe committed:
Remove writing progress notes from Brigand guide

The guide was finished, but I forgot to remove the notes of what was
left to do.

wthrowe committed:
Use lazy list_contains where appropriate

kidder reviewed (dismissed):
Not sure what's going on with clang-tidy...

wthrowe commented:
The redefinition errors are clang-tidy getting confused by the PCH.  If you run it in a no-PCH build they go away.

There are also a bunch of legitimate things like missing closing namespace comments that I will fix.

wthrowe committed:
Fix clang-tidy warnings in TMPL.hpp

wthrowe committed:
Disable PCH in clang-tidy CI jobs

We get clang-tidy errors when a file in the PCH is modified.  I
believe this is caused by `#pragma once` not being respected on the
main file being processed.
